modules/ROOT/pages/_partials/data-sync/server-offline-and-conflict.adoc

```diff
@@ -108,164 +64,104 @@ type Mutation {
 
-[source,javascript]
-----
-const { conflictHandler, strategies } = require('@aerogear/voyager-conflicts')
-----
+.Procedure
 
+. Import the link:https://npmjs.com/package/@aerogear/voyager-conflicts[@aerogear/voyager-conflicts] package.
+
++
 [source,javascript]
 ----
- {
-   updateTask: async (obj, clientData, context, info) => {
-     // 1. Read the Object from the database. This example uses Knex https://knexjs.org/.
-     const task = await context.db('tasks').select().where('id', clientData.id).then((rows) => rows[0])
-
-     // 2. Conflict Detection using `VersionedObjectState`
-     //    If the version number from the database does not match the one sent by the client
-     //    A conflict occurs
-     if (conflictHandler.hasConflict(task, clientData)) {
-       // If a conflict is detected, resolve it on the server using one of the default strategies.
-       const { resolvedState, response } = await conflictHandler.resolveOnServer(strategies.clientWins, task, clientData)
-
-       // persist the resolved data to the database and then return the conflict response
-       await context.db('tasks')
-         .update(resolvedState)
-         .where('id', resolvedState.id)
-         .returning('*')
-         .then((rows) => rows[0])
-
-       return response
-     }
-
-     // 3. Always call nextState before persisting the updated record.
-     // This ensures it has the correct version number
-     conflictHandler.nextState(clientData)
-
-     // 4. Persist the update to the database and return it to the client
-     const update = await context.db('tasks')
-       .update(clientData)
-       .where('id', clientData.id)
-       .returning('*')
-       .then((rows) => rows[0])
-
-     return update
-   }
- }
+const { HashObjectState } = require('@aerogear/voyager-conflicts')
 ----
 
-When there is no conflict, `conflictHandler.nextState(clientData)` is called and the data is persisted. When a conflict occurs, the following happens.
-
-* `conflictHandler.resolveOnServer` is called with the `clientWins` strategy. In this case, the `resolvedState` will be the new data provided by the client. The newly `resolvedState` should be persisted.
-* `conflictHandler.resolveOnServer` also returns a `response`, which should be returned to the client.
-
-The `response` object is a `ConflictResolution` object that tells the client there was a conflict, that it was resolved on the server and provides the new `resolvedState`. In most cases, the client needs to know about conflicts that happen on the server. This allows the client to handle the conflict accordingly. For example, the screen the user is looking at might need to be refreshed with new data after a conflict.
-
-== Conflict Resolution Strategies
-
-There is one default conflict resolution strategy.
+. When using the `HashObjectState` implementation you must provide your own hashing function. The function signature should be as follows:
```

Comment:
. When using the `HashObjectState` implementation, a hashing function must be provided. The function signature should be as follows: